eth/tracers/api.go

```diff
@@ -710,10 +722,26 @@ func (api *API) TraceCall(ctx context.Context, args ethapi.CallArgs, blockNrOrHa
 	}
 	defer release()
 
+	// Apply the customized state rules if required.
+	if config != nil {
+		if err := config.StateOverrides.Apply(statedb); err != nil {
+			return nil, err
+		}
+	}
 	// Execute the trace
 	msg := args.ToMessage(api.backend.RPCGasCap())
 	vmctx := core.NewEVMBlockContext(block.Header(), api.chainContext(ctx), nil)
-	return api.traceTx(ctx, msg, vmctx, statedb, config)
+
+	var traceConfig *TraceConfig
+	if config != nil {
```

Comment:
I had to think a while to figure out why this is needed. So essentially, you do this to clear out the `StateOverrides`, so it doesn't get applied a second time later on? 
If that's the case, why not just set it to `nil`?

Reply:
Because `api.traceTx` only accepts the `TraceConfig`, while `TraceCall` use the `TraceCallConfig`. Here just do the parameter conversion.